### DISCUSSION TITLE
Vm.log throws "Error (7920): Identifier not found or not unique." when testing testPerformUpkeepUpdatesRaffleStateAndEmitsRequestId

### DISCUSSION THREAD

**Discussion by YapBarry** [Cyfrin/foundry-full-course-cu#2104] [Q&A] [answered]
Hi, in Lesson 9, smart contract lottery, I was trying to run forge test on testPerformUpkeepUpdatesRaffleStateAndEmitsRequestId at the end of lesson 35 but it threw the following error. Tried googling and searching the discussions channel but found nothing related to this. Is it because "Vm" is actually an interface in "Vm.sol"?

```
user$ forge test --mt testPerformUpkeepUpdatesRaffleStateAndEmitsRequestId
Compiler run failed:
Error (7920): Identifier not found or not unique.
   --> test/unit/RaffleTest.t.sol:197:9:
    |
197 |         Vm.log[] memory entries = vm.getRecordedLogs();
    |         ^^^^^^

Error: 
Compilation failed
```

**Comment by EngrPips** [comment]
Hello @YapBarry, Please show your test file.

**Comment by Shiva-Sai-ssb** [comment, marked as answer]
Hey @YapBarry, It's a capital "L" in `Vm.Log`.  

Please change the line from:
```solidity
Vm.log[] memory entries = vm.getRecordedLogs();
```
to:
```solidity
Vm.Log[] memory entries = vm.getRecordedLogs();
```

**Reply by EngrPips** [reply]
Good catch

**Reply by YapBarry** [reply]
omg thanks! Cant believe I missed that

**Reply by Shiva-Sai-ssb** [reply]
You're welcome!
Actually, I encountered the same error while I was going through this lesson. That's why I was able to figure out so quickly what the issue was.

**Reply by EngrPips** [reply]
Indeed, That was hard to catch. Over and over, life always teaches us that `experience is the best teacher.`

**Reply by Shiva-Sai-ssb** [reply]
Yes, what you said is absolutely perfect.
